### PR TITLE
added husky pre-commit hook to lint

### PR DESCRIPTION
**Pull request by dsawali**
#1300

Added husky pre-commit hook to lint before a commit
 
Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npm run lint
```
Comment (roxaneletourneau) [inline .husky/pre-commit:4]: Should we run `npm run precommit` instead to apply prettier configuration too?
see https://github.com/ecadlabs/taquito/blob/master/packages/taquito-beacon-wallet/package.json#L37

File: .github/workflows/deploy_edge.yml [hunk 2]
```diff
@@ -23,7 +23,7 @@ jobs:
     - uses: actions/checkout@v2
     - uses: actions/setup-node@v2
       with:
-        node-version: ${{ matrix.node }}
+        node-version: ${{ matrix.nodet }}
```
Comment (roxaneletourneau) [inline .github/workflows/deploy_edge.yml:26]: Can you explain what this change is doing?

File: package.json [hunk 3]
```diff
@@ -15,7 +15,8 @@
     "build-docs": "typedoc",
     "gh-pages": "gh-pages",
     "netlify:preview": "netlify deploy --site $NETLIFY_SITE_ID --auth $NETLIFY_AUTH_TOKEN --json --alias=$short_sha --message 'Deployment for $GITHUB_REF'",
-    "netlify:publish": "netlify deploy --site $NETLIFY_SITE_ID --auth $NETLIFY_AUTH_TOKEN --json --prod --message 'Deployment for $GITHUB_REF'"
+    "netlify:publish": "netlify deploy --site $NETLIFY_SITE_ID --auth $NETLIFY_AUTH_TOKEN --json --prod --message 'Deployment for $GITHUB_REF'",
+    "prepare": ""
```
Comment (roxaneletourneau) [inline package.json:19]: Is the prepare script needed if it is empty?

Comment (dsawali) [reply package.json:19]: it's needed so it doesn't get run in the CI, I was following this part of the docs https://typicode.github.io/husky/#/?id=disable-husky-in-cidockerprod

File: .github/workflows/deploy_edge.yml (hunk 2 above)
Comment (dsawali) [reply .github/workflows/deploy_edge.yml:26]: this might be a mistake, I will fix this

File: package.json (hunk 3 above)
Comment (roxaneletourneau) [reply package.json:19]: got it, thanks!

File: .husky/pre-commit (hunk 1 above)
Comment (dsawali) [reply .husky/pre-commit:4]: sure, that  sounds good. Do you think we should run both lint and precommit? Or will only precommit be suffecient?

Comment (roxaneletourneau) [reply .husky/pre-commit:4]: I think precommit will be sufficient as it will call `eslint --fix`

Comment (dsawali) [reply .husky/pre-commit:4]: gotcha, I'll remove the lint stage then

**Review by roxaneletourneau — APPROVED** [review]

**Pull request merged**